### PR TITLE
WFCORE-3319 - There should be replaced stacktrace loggings to stderr with usage of logging

### PR DESCRIPTION
**Pull request by rpelisse**
Issue: https://issues.jboss.org/browse/JBEAP-12220
Upstream Issue: https://issues.jboss.org/browse/WFCORE-3319

### CODE REVIEW

File: embedded/src/main/java/org/wildfly/core/embedded/EmbeddedStandaloneServerFactory.java [hunk 1]
```diff
@@ -336,10 +336,10 @@ private void exit() {
                 } catch (RuntimeException rte) {
                     throw rte;
                 } catch (InterruptedException ite) {
-                    ite.printStackTrace();
+                    ServerLogger.ROOT_LOGGER.error(ite.getLocalizedMessage(), ite);
```
Comment (bstansberry) [inline embedded/src/main/java/org/wildfly/core/embedded/EmbeddedStandaloneServerFactory.java]: @luck3y I believe these fixes in this file should all use EmbeddedLogger.ROOT_LOGGER, and the uses of ServerLogger in EmbeddedHostControllerFactory should switch to using EmbeddedLogger. (AFAICT the ServerLogger messages used there are not used outside that class so they can just move.) The EmbeddedHostControllerFactory change being a task for you, not for this PR. WDYT?

Comment (rpelisse) [reply embedded/src/main/java/org/wildfly/core/embedded/EmbeddedStandaloneServerFactory.java]: @bstansberry I'm of course OK changing to the EmbeddedLogger. i've poundered about for a little while, not knowing what was best here...

File: io/subsystem/src/main/java/org/wildfly/extension/io/OptionAttributeDefinition.java [hunk 2]
```diff
@@ -137,7 +138,7 @@ private void setType() {
                 }
 
             } catch (Exception e) {
-                e.printStackTrace();
+                IOLogger.ROOT_LOGGER.error(e.getLocalizedMessage(), e);
```
Comment (bstansberry) [inline io/subsystem/src/main/java/org/wildfly/extension/io/OptionAttributeDefinition.java]: This should catch the specific checked exceptions the code can throw, wrap them in RuntimeException and rethrow. We shouldn't be swallowing problems here.

File: patching/src/main/java/org/jboss/as/patching/tool/PatchingHistory.java [hunk 3]
```diff
@@ -426,7 +426,7 @@ public Patch getMetadata() {
                                     try {
                                         patch = ((PatchBuilder)PatchXml.parse(patchXml)).build();
                                     } catch (Exception e) {
-                                        e.printStackTrace(); // TODO
+                                        PatchLogger.ROOT_LOGGER.error(e.getLocalizedMessage(), e);
```
Comment (bstansberry) [inline patching/src/main/java/org/jboss/as/patching/tool/PatchingHistory.java]: Please leave the // TODO bit unless @aloubyansky has decided he doesn't want it any more.

**Comment by bstansberry** [comment]
Thanks for this!

File: patching/src/main/java/org/jboss/as/patching/tool/PatchingHistory.java (hunk 3 above)
Comment (aloubyansky) [reply patching/src/main/java/org/jboss/as/patching/tool/PatchingHistory.java]: You can remove the TODO, thanks!

File: io/subsystem/src/main/java/org/wildfly/extension/io/OptionAttributeDefinition.java (hunk 2 above)
Comment (rpelisse) [reply io/subsystem/src/main/java/org/wildfly/extension/io/OptionAttributeDefinition.java]: @bstansberry I utterly agree with that, but here I just wanted to maintain the current behavior. If we know rethrow exception, its does mean that "things that work but printed stacktraces" are now going to "fails" (or at least throw exception). 

For Wildfly, as the next release is 11..0.0. that seems OK to do so, and I will update my PR right away, but I wonder if we should have the same strategy for EAP 7. (see https://github.com/jbossas/jboss-eap7/pull/2426)

File: embedded/src/main/java/org/wildfly/core/embedded/EmbeddedStandaloneServerFactory.java (hunk 1 above)
Comment (bstansberry) [reply embedded/src/main/java/org/wildfly/core/embedded/EmbeddedStandaloneServerFactory.java]: @rpelisse ok, let's go with EmbeddedLogger. Thanks!

**Comment by luck3y** [comment]
@bstansberry Sorry I'm late to this, I totally agree though, these shouldn't just catch and print, if there are failures due to that they are separate issues / bugs. EmbeddedLogger.ROOT_LOGGER sounds fine to me.

@rpelisse, thanks for noticing those, and for fixing!

**Comment by rpelisse** [comment]
Last Week
====

1) EAP 7
* https://issues.jboss.org/browse/JBEAP-8937 - build.sh and integration-tests.sh scripts doesn't work on Solaris SPARC (WIP)
** issue has been closed for 7.0 and 7.1 and but pushed to 7.2: https://issues.jboss.org/browse/JBEAP-13246 / PR: https://github.com/jbossas/jboss-eap7/pull/2430
* https://issues.jboss.org/browse/JBEAP-12242 - NPE in io.undertow.client (WIP)
** Resume work on reproducing the environement (SLL part)
* https://issues.jboss.org/browse/JBEAP-12220 - There should be replaced stacktrace loggings to stderr with usage of logging.
** did an inventory, issue boils down to only a few occurences, spread across Widfly and Wildfly-core.
** Wildfly - https://issues.jboss.org/browse/JBEAP-12220 / PR: https://github.com/jbossas/jboss-eap7/pull/2426
** Wildfly Core - https://issues.jboss.org/browse/WFCORE-3319 / PR: https://github.com/wildfly/wildfly-core/pull/2848
* https://issues.jboss.org/browse/JBEAP-9140 - Improve buffer-pool model description
** As peer Brian's comment: https://github.com/wildfly/wildfly-core/pull/2400
** needs to be reworked to use a transformer to avoid breaking compat

2) RPM EAP / PROD and BRMS
* RPM 201: http://ftp.rpm.org/max-rpm/ - finished reading up and released my notes: https://mojo.redhat.com/groups/jboss-sustaining-engineering-team/blog/2017/09/29/max-rpm-cliff-notes

3) Bugclerk
A) Review and merge Bart's check request:
* Check Request - https://github.com/jboss-set/bug-clerk/issues/74 - split it into two check requests
** https://github.com/jboss-set/bug-clerk/issues/76 - Check Request #33 - Only issues with QA ack can be worked on (DONE)
** https://github.com/jboss-set/bug-clerk/issues/75 - Check Request #32 - All issues on payload should have 3 acks (WIP)
*** This one needs to be fixed, as the current logic works on *any* issue, not on the one associated with a payload
* https://github.com/jboss-set/bug-clerk/issues/45 - Fix Version white list 
** this is going nowhere because we lack a way of listing the "who can", firing an email to start the discussion
B) Maven Plugin for Bugclerk
* fiddle a bit to find a way to pass custom checks to the plugin, add user documentation and release plugin
C) Start refactoring work to have Bugclerk placed "behind" Aphrodite
* create a checks component (Janitor)

4) Thunder - Storage Issue on / and /boot
* /boot warning was a false+ fixed it: https://github.com/jboss-set/zeus/commit/551f59213c970784c9edb0ff453aa5a3a6c52414
* logfiles from EAP are consuming too much space:
```xml
 			<periodic-size-rotating-file-handler name="FILE" autoflush="true">
                <formatter>
                    <named-formatter name="PATTERN"/>
                </formatter>
                <file relative-to="jboss.server.log.dir" path="server.log"/>
                <suffix value=".yyyy-MM-dd"/>
                <append value="true"/>
                <max-backup-index value="31"/>
                <rotate-size value="1G"/>
            </periodic-size-rotating-file-handler>
```


Misc:
* JBoss Weekly Editorial on Thursday - https://developer.jboss.org/people/rpelisse/blog/2017/09/29/this-week-in-jboss-28-sept-2017-to-bodly-go-where-to-javaone-has-not-gone-before
* Released blog entry on git-worktree that I've been sitting for weeks - https://mojo.redhat.com/groups/jboss-sustaining-engineering-team/blog/2017/09/29/set-engineering-practise-git-worktree
* As mentioned above, this an other blog entry about my RPM readings - https://mojo.redhat.com/groups/jboss-sustaining-engineering-team/blog/2017/09/29/max-rpm-cliff-notes

This Week
====

Tuesday is Public Holiday in Germany (German Unification day) so I won't be in the call tomorrow.

1) EAP 7
* https://issues.jboss.org/browse/JBEAP-12242 - NPE in io.undertow.client (WIP)
** Resume work on reproducing the environement (SLL part)
* https://issues.jboss.org/browse/JBEAP-12220 - There should be replaced stacktrace loggings to stderr with usage of logging.
** Wildfly Core - https://issues.jboss.org/browse/WFCORE-3319 / PR: https://github.com/wildfly/wildfly-core/pull/2848 (some follow up work to be done)
* https://issues.jboss.org/browse/JBEAP-9140 - Improve buffer-pool model description
** As peer Brian's comment: https://github.com/wildfly/wildfly-core/pull/2400
** needs to be reworked to use a transformer to avoid breaking compat

2) Bugclerk

A) Review and merge Bart's check request:
* https://github.com/jboss-set/bug-clerk/issues/45 - Fix Version white list 
** see if I can implements Carlo's suggestion
B) Maven Plugin for Bugclerk
* fiddle a bit to find a way to pass custom checks to the plugin, add user documentation and release plugin
C) Start refactoring work to have Bugclerk placed "behind" Aphrodite
* create a checks component (Janitor)

**Comment by bstansberry** [comment]
@rpelisse Sorry, I just saw your question re: my comment on the change to OptionAttributeDefinition. This kind of decision is really a case-by-case thing. In the OptionAttributeDefinition.Builder case it's internal code (not supported for other use) that's only called from a fixed set of places. And if any exception occurred there, the software would be broken and would fail in some other fashion, probably a really mysterious one if the stderr logging wasn't noticed. So throwing the exception is better.

**Comment by rpelisse** [comment]
@bstansberry no worries, I've been lost in a PR flow more than once. Anyway, I've opted to throw the exception anyway, as the next version will be "major" enough to justify such change of behavior. In a *.*.z stream, one could have argue it could have been ill-advised, but you are right anyway, it's internal and better anyway.

@luck3y credit for raisong the issue goes to the reporter @rhatlapa - I just indulged my love of using 'grep' recursively...

_To anybody following this PR: As far as I know I've integrated all the changes requested, so on my side it is "ready to go". Please let me know I somehow forgot something..._

File: process-controller/src/main/java/org/jboss/as/process/ManagedProcess.java [hunk 4]
```diff
@@ -200,7 +200,6 @@ void doStart(boolean restart) {
         try {
             process = builder.start();
         } catch (IOException e) {
-            e.printStackTrace();
             processController.operationFailed(processName, ProcessMessageHandler.OperationType.START);
             log.failedToStartProcess(processName);
```
Comment (rhatlapa) [inline process-controller/src/main/java/org/jboss/as/process/ManagedProcess.java]: Having info why the process failed to start might be useful => shouldn't the info from the IOException stay logged somewhere?

**Comment by rpelisse** [comment]
retest this please

File: process-controller/src/main/java/org/jboss/as/process/logging/ProcessLogger.java [hunk 5]
```diff
@@ -820,4 +820,12 @@
...
     void destroyingProcess(String process, long timeout);
 
+    /**
+     * Logs an error message indicating the process failed to start, along with exception associated to the failure.
+     *
+     * @param processName the process name.
+     */
+    @LogMessage(level = ERROR)
+    @Message(id = 9, value = "Failed to start process '%s': %s")
+    void failedToStartProcess(String processName, Throwable t);
```
Comment (bstansberry) [inline process-controller/src/main/java/org/jboss/as/process/logging/ProcessLogger.java]: This won't log a stack trace. Please change the existing method to

```
    @LogMessage(level = ERROR)
    @Message(id = 9, value = "Failed to start process '%s'")
    void failedToStartProcess(@Cause Throwable cause, String processName);
```

**Comment by rpelisse** [comment]
@bstansberry done! Thanks for pointing this out!

**Comment by rpelisse** [comment]
retest this please

File: process-controller/src/main/java/org/jboss/as/process/logging/ProcessLogger.java (hunk 5 above)
Comment (rpelisse) [reply process-controller/src/main/java/org/jboss/as/process/logging/ProcessLogger.java]: @bstansberry fyi this is now fixed.

**Comment by bstansberry** [comment]
Thanks!

**Pull request merged**